fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -391,13 +393,33 @@ void SpecialKeySpace::clear(ReadYourWritesTransaction* ryw, const KeyRef& key) {
 	return impl->clear(ryw, key);
 }
 
+bool validateSnakeCaseNaming(const KeyRef& k) {
+	KeyRef key(k);
+	// Remove prefix \xff\xff
+	ASSERT(key.startsWith(specialKeys.begin));
+	key = key.removePrefix(specialKeys.begin);
+	// Suffix can be \xff\xff or \x00 in single key range
+	if (key.endsWith(specialKeys.begin))
+		key = key.removeSuffix(specialKeys.end);
+	else if (key.endsWith(LiteralStringRef("\x00")))
+		key = key.removeSuffix(LiteralStringRef("\x00"));
+	for (const char& c : key.toString()) {
+		// only small letters, numbers, '/', '_' is allowed
+		ASSERT((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || c == '/' || c == '_');
+	}
+	return true;
+}
+
 void SpecialKeySpace::registerKeyRange(SpecialKeySpace::MODULE module, SpecialKeySpace::IMPLTYPE type,
                                        const KeyRangeRef& kr, SpecialKeyRangeReadImpl* impl) {
 	// module boundary check
-	if (module == SpecialKeySpace::MODULE::TESTONLY)
+	if (module == SpecialKeySpace::MODULE::TESTONLY) {
 		ASSERT(normalKeys.contains(kr));
-	else
+	} else {
 		ASSERT(moduleToBoundary.at(module).contains(kr));
+		ASSERT(validateSnakeCaseNaming(kr.begin) &&
+		       validateSnakeCaseNaming(kr.end)); // validate keys follow snake case naming style
```

Comment:
I supposed one could imagine a prefix range where the beginning satisfies this predicate but the end doesn't. That said, I think we want all our prefix ranges to end in `/`, so I think this is ok.

Reply:
Not sure I understand this correctly.
The checker right now only guarantees the prefix key is in small letters, numbers, `/` and `_`.
If a range is like (`\xff\xff/prefix/`, `\xff\xff/prefix0` or `\xff\xff/prefix\xff\xff`), then it's allowed.